aries/agents/mediator-client/src/mediator_client.rs

```diff
@@ -0,0 +1,23 @@
+use messages::msg_fields::protocols::out_of_band::invitation::Invitation as OOBInvitation;
+use reqwest::{self, Url};
+
+use crate::errors::MediatorClientResult;
+pub struct MediatorClient {
+    url: Url,
+    client: reqwest::Client,
+}
+
+impl MediatorClient {
+    pub fn new(url: &str) -> MediatorClientResult<Self> {
+        Ok(Self {
+            url: url.parse()?,
+            client: reqwest::Client::new(),
+        })
+    }
+
+    pub async fn register(&self) -> MediatorClientResult<OOBInvitation> {
+        let endpoint = self.url.join("/register.json")?;
```

Comment:
Out of PR scope, just another task note: on mediator, lets:
-  change `/register.json` to `/invitation`
- remove existing `/register` - this won't be used in practice, and if we really want to, we can always render QRs in client's browser.
   - therefore also remove the `qr` related dependencies from the project